dbms/src/Storages/MergeTree/KeyCondition.cpp

```diff
@@ -1095,17 +1097,23 @@ bool KeyCondition::mayBeTrueInParallelogram(const std::vector<Range> & parallelo
         }
         else if (element.function == RPNElement::FUNCTION_NOT)
         {
+            assert(!rpn_stack.empty());
```

Comment:
Note that assert will be used only in debug build.
But in debug build we use debug version of libc++ that already contains that assert inside std::vector::back.

This assert is needed to provide info for compiler to avoid a warning.